components/style_derive/cg.rs

```diff
@@ -136,6 +137,16 @@ where
     }
 }
 
+pub fn parse_variant_attrs<A>(variant: &Variant) -> A
+where
+    A: FromVariant,
+{
+    match A::from_variant(variant) {
+        Ok(attrs) => attrs,
+        Err(e) => panic!("failed to parse attributes: {}", e),
```

Comment:
With `expect`:

```
    = help: message: failed to parse attributes: Error { kind: Multiple([Error { kind: MissingField("function"), locations: [] }, Error { kind: MissingField("comma"), locations: [] }]), locations: []
```

With an explicit panic and `Display`:

```
    = help: message: failed to parse attributes: Missing field `comma
```

The actual code if it was important for performances should be:

```rust
#[inline(never)]
#[cold]
fn panic(e: Error) -> ! {
    panic!("failed to parse attributes: {}", e)
}

match A::from_variant(variant) {
    Ok(attrs) => attrs,
    Err(e) => panic(e),
}
```